### ISSUE TITLE
Release Intel Arc GPU Gold support for Geti model training

### ISSUE THREAD

**Issue by BrittvanAkker** [open-edge-platform/geti_v2#714]
**User story**

To align Intel software adoption with Intel hardware enabling, Intel software should be able to run on Intel hardware so that customers are able to start and complete their AI journey on an Intel-based silicon stack. To bridge this gap with Geti, Beta support for model fine-tuning on Intel Arc GPU (Battlemage 12GB & 24GB) has been released in May 2025 (v2.9). 

The objective of this requirement is to achieve Gold Support for Intel Arc GPU by fixing the last open issues for any remaining architectures that didn’t perform on-par yet in the Beta release. 
 
**Requirements**

- Geti shall be able to utilize Intel GPUs for on-prem training across the whole Geti model suite.
   - Target Intel GPU: Intel Arc (Battlemage).
   - Model performance shall be on-par with the RTX 3090 as a reference card.

- Performance parity with the reference card shall be achieved with the remaining model architectures that did not perform on-par yet as part of the Beta Support release (May, v2.9). 
   - The accuracy and training speed for detection models shall be improved.

- Benchmarking shall continue to ensure performance parity on current and future algorithms.

- Internal and external documentation shall be updated accordingly.

- Multi-vendor (Intel + other) GPU systems will not be expected to be officially supported at this time.

**Comment by leoll2** [comment]
As of 2.12, all the models should be trainable on XPU. There are still some open issues on IPEX side concerning the training speed of specific models architectures (D-Fine, RTDetr, MaskRCNN).
- https://jira.devtools.intel.com/browse/PYTORCHDGQ-5687
- https://jira.devtools.intel.com/browse/PYTORCHDGQ-5671
